src/test/scala/mesosphere/marathon/api/v2/validation/PodsValidationTest.scala

```diff
@@ -124,6 +125,96 @@ class PodsValidationTest extends UnitTest with ValidationTestLike with PodsValid
     }
   }
 
+  "with persistent volumes" should {
+    "be valid" in new Fixture {
+      val pod = validResidentPod
+      validator(pod) should be(aSuccess)
+    }
+
+    "be valid if no unreachable strategy is provided" in new Fixture {
+      val pod = validResidentPod.copy(scheduling = validResidentPod.scheduling.map(_.copy(
+        unreachableStrategy = None)))
+      validator(pod) should be(aSuccess)
+    }
+
+    "be valid if no upgrade strategy is provided" in new Fixture {
+      val pod = validResidentPod.copy(scheduling = validResidentPod.scheduling.map(_.copy(
+        upgrade = None)))
+      validator(pod) should be(aSuccess)
+    }
+
+    "be invalid if unreachable strategy is enabled" in new Fixture {
+      val pod = validResidentPod.copy(scheduling = validResidentPod.scheduling.map(_.copy(
+        unreachableStrategy = Some(UnreachableEnabled()))))
+      validator(pod) should haveViolations(
+        "/" -> "unreachableStrategy must be disabled for pods with persistent volumes")
+    }
+
+    "be invalid if upgrade strategy has maximumOverCapacity set to non-zero" in new Fixture {
+      val pod = validResidentPod.copy(scheduling = validResidentPod.scheduling.map(_.copy(
+        upgrade = Some(PodUpgradeStrategy(maximumOverCapacity = 0.1)))))
+      validator(pod) should haveViolations(
+        "/upgrade/maximumOverCapacity" -> "got 0.1, expected 0.0")
+    }
+
+    "be invalid if cpu changes" in new Fixture {
+      val pod = validResidentPod.fromRaml
+      val to = pod.copy(containers = pod.containers.map(ct => ct.copy(resources = ct.resources.copy(cpus = 3))))
+      residentUpdateIsValid(pod)(to) should haveViolations(
+        "/" -> "cpus cannot be updated if a pod has persistent volumes")
```

Comment:
Should we extract of these error messages to a constant? (We have tended to do this elsewhere)